test/unit/puppetlabs/services/master/master_core_test.clj

```diff
@@ -189,6 +190,31 @@
                         "supported"]
                        "bogus"}]})))))))
 
+(deftest valid-static-file-path-test
+  (let [valid-paths ["modules/foo/files/bar.txt"
+                     "modules/foo/files/bar"
+                     "modules/foo/files/bar/baz.txt"
+                     "modules/foo/files/bar/more/path/elements/baz.txt"]
+        invalid-paths ["modules/foo/manifests/bar.pp"
+                       "manifests/site.pp"
+                       "environments/foo/bar/files"
+                       "environments/../manifests/files/site.pp"
+                       "environments/../modules/foo/lib/puppet/parser/functions/site.rb"
+                       "environments/production/files/~/.bash_profile"
+                       "environments/../modules/foo/files/site.pp"
+                       "environments/production/modules/foo/files/../../../../../../site.pp"
+                       "environments/production/modules/foo/files/bar.txt"
+                       "environments/production/modules/foo/files/..conf..d../..bar..txt.."
+                       "environments/test/modules/foo/files/bar/baz.txt"
+                       "environments/dev/modules/foo/files/path/to/file/something.txt"]
```

Comment:
The following manifest is totally okay in puppet, so I don't think rejecting %2E outright would be great. It would potentially disallow valid requests. I don't have a very good sense of what file names are in use for file resources though.

```
file {"/tmp/foo":
  ensure => present,
  source => 'puppet:///modules/foo/bar..d/foo%2Etxt',
}
```